src/main/java/com/google/sps/data/LearningPathService.java

```diff
@@ -198,6 +252,17 @@ public LearningItem submitFeedback(long pathId, long learningItemId, String user
 
 			existing.setRating(rating);
 			existing.setCompleted(completed);
+
+            Entity feedback = new Entity(ITEM_FEEDBACK);
```

Comment:
We need to set the existing itemId here, or datastore will create a new (duplicate) ItemFeedback object here.

I'm pushing a  fix for this to the branch.